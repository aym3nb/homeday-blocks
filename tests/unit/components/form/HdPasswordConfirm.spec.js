import { shallowMount } from '@vue/test-utils';
import HdPasswordConfirm from '@/components/form/HdPasswordConfirm.vue';
import HdInputPassword from '@/components/form/HdInputPassword.vue';

const STRENGTH_METER_SELECTOR = '.confirmPassword__strengthMeter';

describe('HdPasswordConfirm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(HdPasswordConfirm);
  });

  it('is rendered as expected', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('has two HdInputPassword inputs', () => {
    expect(wrapper.findAll(HdInputPassword).length).toEqual(2);
  });

  it('has a password strength meter', () => {
    expect(wrapper.find(STRENGTH_METER_SELECTOR).exists()).toBeTruthy();
  });
});
