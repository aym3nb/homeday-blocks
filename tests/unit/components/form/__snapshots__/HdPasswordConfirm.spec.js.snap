// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdPasswordConfirm is rendered as expected 1`] = `
<div class="confirmPassword">
  <hdinputpassword-stub value="" name="passwordMain" icon="" label="Ihr Passwort" required="true" class="confirm-password__input"></hdinputpassword-stub>
  <div class="confirmPassword__strengthMeter">
    <div class="confirmPassword__strengthMeter__text"></div>
    <div class="confirmPassword__strengthMeter__gauge">
      <div class="confirmPassword__strengthMeter__gauge__bar"></div>
    </div>
  </div>
  <hdinputpassword-stub value="" name="passwordConfirm" icon="" label="BestÃ¤tigen Sie Ihr Passwort" required="true" class="confirmPassword__input"></hdinputpassword-stub>
</div>
`;
